src/Xunit.Combinatorial/CombinatorialMemberDataAttribute.cs

```diff
@@ -75,19 +82,10 @@ public CombinatorialMemberDataAttribute(string memberName, params object?[]? arg
     /// <returns>The generic type argument for (one of) the <see cref="IEnumerable{T}"/> interface)s) implemented by the <paramref name="enumerableType"/>.</returns>
     private static TypeInfo? GetEnumeratedType(Type enumerableType)
     {
-        if (enumerableType.IsGenericType)
+        if (enumerableType.IsGenericType && enumerableType.GetGenericTypeDefinition() == typeof(IEnumerable<>))
```

Comment:
Does this get rid of support for `TheoryData<>`?

Reply:
No, it shouldn't. `TheoryData<T>` implements `IEnumerable<T>,` so this should still be compatible.
The previous implementation had two identical nested conditions, but since there is no need to explicitly test for `TheoryData<T>`, I merged it into a single one.